### PR TITLE
Fix mypy errors (dagcircuit)

### PR DESCRIPTION
**Pull request by Randl**
### Summary

Following discussion, I'm splitting https://github.com/Qiskit/qiskit-terra/pull/8187 by module.

### Details and comments

This fixes the only mypy errors in dagcircuit as for now.

### CODE REVIEW

File: qiskit/dagcircuit/dagnode.py [hunk 1]
```diff
@@ -100,8 +103,8 @@ def __init__(self, op, qargs=(), cargs=()):
         """Create an Instruction node"""
         super().__init__()
         self.op = op
-        self.qargs = tuple(qargs)
-        self.cargs = tuple(cargs)
+        self.qargs: Optional[Tuple[Qubit, ...]] = tuple(qargs)
+        self.cargs: Optional[Tuple[Clbit, ...]] = tuple(cargs)
```
Comment (1ucian0) [inline qiskit/dagcircuit/dagnode.py:107]: not an expert in type annotation. Would it be better to add them into the method declaration?

**Review by 1ucian0 — COMMENTED** [review]
thanks for splitting it! I finally got some time to have a look at this one.

File: qiskit/dagcircuit/dagnode.py (hunk 1 above)
Comment (Randl) [reply qiskit/dagcircuit/dagnode.py]: yes, you're right; I've changed it

**Comment by Randl** [comment]
@1ucian0 can you approve, please? :)

**Review by 1ucian0 — APPROVED** [review]
Thanks for your patience and your work!

**Pull request merged**